ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -221,6 +231,28 @@ private static PhysicalSchema getPhysicalSchemaWithKeyAndPseudoCols(
     );
   }
 
+  private static PhysicalSchema getPhysicalSchemaWithPseudoColumnsToMaterialize(
+      final SourceStep<?> streamSource) {
+
+    FormatInfo f = streamSource.getFormats().getKeyFormat();
+    SerdeFeatures s = streamSource.getFormats().getKeyFeatures();
+    KeyFormat k = streamSource instanceof WindowedTableSource
+        ? KeyFormat.windowed(f, s, ((WindowedTableSource) streamSource).getWindowInfo())
+        : KeyFormat.nonWindowed(f, s);
```

Comment:
This should, in it's current state, never be called on `WindowedTableSource`s. While it's probably a bug that it isn't, [considering joins on windowed table sources don't even work](https://github.com/confluentinc/ksql/issues/8055), I think I should remove this for now to reduce code complexity. WDYT @vcrfxia?